usersync/ejector.go

```diff
@@ -0,0 +1,118 @@
+package usersync
+
+import (
+	"math"
+	"time"
+)
+
+type Ejector interface {
+	Choose(uids map[string]UIDEntry) (string, error)
+}
+
+type OldestEjector struct{}
+
+type PriorityBidderEjector struct {
+	PriorityGroups  [][]string
+	syncersByBidder map[string]Syncer
+	OldestEjector   OldestEjector
+}
+
+// Choose method for oldest ejector will return the oldest uid
+func (o *OldestEjector) Choose(uids map[string]UIDEntry) (string, error) {
+	var oldestElement string
+	var oldestDate time.Duration = math.MaxInt64
+
+	for key, value := range uids {
+		timeUntilExpiration := time.Until(value.Expires)
```

Comment:
Small optimization: `time.Until(value.Expires)` calls `time.Now()` under the hood at every loop. We can both simplify and optimize if we use the `Before(u Time) bool` function instead
```
20   // Choose method for oldest ejector will return the oldest uid
21   func (o *OldestEjector) Choose(uids map[string]UIDEntry) (string, error) {
22       var oldestElement string
23 -     var oldestDate time.Duration = math.MaxInt64
   +     var oldestDate time.Time = time.Unix(1<<63-62135596801, 999999999) // Max value for time
24   
25       for key, value := range uids {
26 -         timeUntilExpiration := time.Until(value.Expires)
27 -         if timeUntilExpiration < time.Duration(oldestDate) {
   +         if value.Expires.Before(oldestDate) {
28               oldestElement = key
29               oldestDate = timeUntilExpiration
30           }
31       }
32       return oldestElement, nil
33   }
usersync/ejector.go
```